### ISSUE TITLE
configure: error: When compiling, the header files cannot be found.

### ISSUE THREAD

**Issue by GotoRen** [smashedtoatoms/asdf-postgres#71]
**Describe the bug**

When trying to install "postgres@15.2" via asdf, an error occurs.

I prepared a script similar to the following to automate the installation of multiple plugin tools.

- `install-tools.sh`

```shell
#!/bin/bash -u

set -o nounset

tools="$(
  cd $(dirname $0)
  pwd
)/../../.tool-versions"

awk '{ print $1; }' ${tools} | while read tool; do
  asdf plugin add ${tool}
done

asdf install
```

```
configure: using LDFLAGS=-isysroot $(PG_SYSROOT)   -L/usr/local/opt/openssl/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/openssl@3/lib -L/usr/local/opt/openssl@1.1/lib -L/usr/local/lib -L/usr/local/lib -Wl,-dead_strip_dylibs
configure: creating ./config.status
config.status: creating GNUmakefile
config.status: creating src/Makefile.global
config.status: creating src/include/pg_config.h
config.status: src/include/pg_config.h is unchanged
config.status: creating src/include/pg_config_ext.h
config.status: src/include/pg_config_ext.h is unchanged
config.status: creating src/interfaces/ecpg/include/ecpg_config.h
config.status: src/interfaces/ecpg/include/ecpg_config.h is unchanged
config.status: linking src/backend/port/tas/dummy.s to src/backend/port/tas.s
config.status: linking src/backend/port/sysv_sema.c to src/backend/port/pg_sema.c
config.status: linking src/backend/port/sysv_shmem.c to src/backend/port/pg_shmem.c
config.status: linking src/include/port/darwin.h to src/include/pg_config_os.h
config.status: linking src/makefiles/Makefile.darwin to src/Makefile.port
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src all
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C common all
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Werror=unguarded-availability-new -Wendif-labels -Wmissing-format-attribute -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -Wno-unused-command-line-argument -Wno-compound-token-split-by-macro -O2 -DFRONTEND -I. -I../../src/common -I../../src/include  -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk   -I/usr/local/opt/openssl/include -I/usr/local/opt/openssl/include -I/usr/local/opt/openssl@3/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/include -I/usr/local/include -I/usr/local/lib -DVAL_CC="\"gcc\"" -DVAL_CPPFLAGS="\"-isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk -I/usr/local/opt/openssl/include -I/usr/local/opt/openssl/include -I/usr/local/opt/openssl@3/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/include -I/usr/local/include -I/usr/local/lib\"" -DVAL_CFLAGS="\"-Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Werror=unguarded-availability-new -Wendif-labels -Wmissing-format-attribute -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -Wno-unused-command-line-argument -Wno-compound-token-split-by-macro -O2\"" -DVAL_CFLAGS_SL="\"\"" -DVAL_LDFLAGS="\"-isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk -L/usr/local/opt/openssl/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/openssl@3/lib -L/usr/local/opt/openssl@1.1/lib -L/usr/local/lib -L/usr/local/lib -Wl,-dead_strip_dylibs\"" -DVAL_LDFLAGS_EX="\"\"" -DVAL_LDFLAGS_SL="\"\"" -DVAL_LIBS="\"-lpgcommon -lpgport -lssl -lcrypto -lz -lreadline -lm \""  -c -o hashfn.o hashfn.c
In file included from hashfn.c:24:
In file included from ../../src/include/postgres.h:47:
../../src/include/utils/elog.h:73:10: fatal error: 'utils/errcodes.h' file not found
#include "utils/errcodes.h"
         ^~~~~~~~~~~~~~~~~~
1 error generated.
make[3]: *** [hashfn.o] Error 1
make[2]: *** [all-common-recurse] Error 2
make[1]: *** [all-src-recurse] Error 2
make: *** [install-tools] Error 1
```

**Steps to reproduce**

1. First, prepare the following directory structure:

```shell
.
├── Makefile
├── .tool-versions
└── scripts
       └── make-rules
               └── install-tools.sh
```

2. The, add "postgres 15.2" to the `.tool-versions`.

```
postgres 15.2
```

3. Finally, execute the above script file through the `make` command.

```Makefile
.PHONY: install-tools
install-tools: ## Install the asdf plugin
	@scripts/make-rules/install-tools.sh
```

```shell
$ make install-tools
```

As a result, you get the error message mentioned above.

**Expected behavior**

The prepared script is a simple one that installs plugins based on the `.tool-versions`.

Therefore, the postgres tool should be installed as expected.

**Screenshots**

<img width="1431" alt="スクリーンショット 2023-05-17 15 48 01" src="https://github.com/smashedtoatoms/asdf-postgres/assets/63791288/2e0777f2-50fe-4319-b6b5-553b58473ef0">

**Additional context**

- macOS 13.3.1 (a)
- Intel Chip (x86_64) and Apple silicon (ARMv8)

- source
  - https://github.com/asdf-vm/asdf-plugins/blob/master/plugins/postgres

**Comment by GotoRen** [comment]
I am waiting for the developer's response to this question.

**Comment by GotoRen** [comment]
Thank you for the reference link.

> I'd guess that Make is what is screwing it up. Make tends to tromp on environment variables when you call makefiles from other makefiles. 

I also strongly agree with this opinion. Indeed, it is not desirable to call a Makefile from another Makefile using a script.

**Comment by smashedtoatoms** [comment]
@GotoRen, I hope @ulope's suggestion helps.  I am going to close this because I can't fix this level of problem for this library.  I made the unfortunate decision to compile from source when I created this. Since every OS has its bespoke linking strategy, and users come up with their own to deal with incompatibilities, I can't keep up.  It has the benefit of making it work in a lot of scenarios, but it also means the user is going to be responsible for chasing down build scenarios any time the environment strays very far from MacOS and Ubuntu defaults.  I'm really sorry.
